src/Network/Network/NetworkWatcher/ConnectionMonitor/SetAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -79,178 +102,404 @@ public class SetAzureNetworkWatcherConnectionMonitorCommand : ConnectionMonitorB
+            ParameterSetName = "SetByResourceV2")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Notes associated with connection monitor.",
+            ParameterSetName = "SetByNameV2")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Notes associated with connection monitor.",
+            ParameterSetName = "SetByLocationV2")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "ConnectionMonitor resource ID.",
+            ParameterSetName = "SetByResourceIdV2")]
+        public string Note { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Configure connection monitor, but do not start it.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Configure connection monitor, but do not start it.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Configure connection monitor, but do not start it.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "ConnectionMonitor resource ID.",
+            ParameterSetName = "SetByResourceId")]
         [ValidateNotNullOrEmpty]
         public SwitchParameter ConfigureOnly { get; set; }
 
         [Parameter(
             Mandatory = false,
-            HelpMessage = "A hashtable which represents resource tags.")]
+            HelpMessage = "A hashtable which represents resource tags.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.",
+            ParameterSetName = "SetByResourceV2")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.",
+            ParameterSetName = "SetByNameV2")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.",
+            ParameterSetName = "SetByLocationV2")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "ConnectionMonitor resource ID.",
+            ParameterSetName = "SetByResourceId")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "ConnectionMonitor resource ID.",
+            ParameterSetName = "SetByResourceIdV2")]
         public Hashtable Tag { get; set; }
 
+        [Parameter(
```

Comment:
Do not add a Force parameter that you are not using

Reply:
removed